controllers/client/grafana_client.go

```diff
@@ -124,21 +126,55 @@ func getAdminCredentials(ctx context.Context, c client.Client, grafana *v1beta1.
 	return credentials, nil
 }
 
-func NewHTTPClient(grafana *v1beta1.Grafana) *http.Client {
-	var timeout time.Duration
-	if grafana.Spec.Client != nil && grafana.Spec.Client.TimeoutSeconds != nil {
-		timeout = time.Duration(*grafana.Spec.Client.TimeoutSeconds)
-		if timeout < 0 {
-			timeout = 0
+// build the tls.Config object based on the content of the Grafana CR object
+func buildTLSConfiguration(ctx context.Context, c client.Client, grafana *v1beta1.Grafana) (*tls.Config, error) {
```

Comment:
I'd move this to the new `http_client.go` file

Reply:
I have choose to move it in a dedicated tls.go file in the same package. Doing this, all element related with tls should be accessible here.